server/player/provider.go

```diff
@@ -0,0 +1,32 @@
+package player
+
+import "github.com/google/uuid"
+
+// Provider represents a value that may provide data to a Player value. It usually does the reading and
+// writing of the player data so that the Player may use it.
+type Provider interface {
+	// Save is called when the player leaves the server and passes on the current player data.
+	Save(UUID uuid.UUID, data Data) error
+	// Load is called when the player joins and passes the UUID of the player.
+	// It expects to the player data, and a bool that indicates whether or not the player has played before.
+	// If this bool is false the player will use default values and you can use an empty Data struct.
+	Load(UUID uuid.UUID) (Data, error)
+	// Close is called when the server closes and is useful to safely close your database.
+	Close()
```

Comment:
Would return an error here to implement `io.Closer` (or you can embed `io.Closer` to do that)